arcstat: add 'avail', fix 'free'

ahrens opened the pull request:
<!--- Please fill out the following template, which will help other contributors review your Pull Request. -->

<!--- Provide a general summary of your changes in the Title above -->

<!---
Documentation on ZFS Buildbot options can be found at
https://openzfs.github.io/openzfs-docs/Developer%20Resources/Buildbot%20Options.html
-->

### Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
The meaning of the `free` field is currently `zfs_arc_sys_free`, which
is the target amount of memory to leave free for the system, and is
constant after booting.

### Description
<!--- Describe your changes in detail -->
This commit changes the meaning of `free` to arc_free_memory(), the
amount of memory that the ARC considers to be free.

It also adds a new arcstat field `avail`, which tracks
`arc_available_memory()`.  This field is printed with the default output.

Since `avail` can be negative, it also updates the arcstat script to
pretty-print negative values.

Add a field `size`, which is the same as `arcsz`.  This matches the implementation (and `arc` is redundant since everything in `arcstat` relates to the ARC).  `size` is displayed by default, rather than `arcsz`, but for backwards compatibility, `arcsz` can still be specified with `-f`.

### How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
<!--- If your change is a performance enhancement, please provide benchmarks here. -->
<!--- Please think about using the draft PR feature if appropriate -->
example output:
```
  $ arcstat -f time,miss,arcsz,c,grow,need,free,avail 1
      time  miss  arcsz     c  grow  need  free  avail
  15:03:02   39K   114G  114G     0     0  2.4G  407M
  15:03:03   42K   114G  114G     0     0  2.1G  120M
  15:03:04   40K   114G  114G     0     0  1.8G  -177M
  15:03:05   24K   113G  112G     0     0  1.7G  -269M
  15:03:06   29K   111G  110G     0     0  1.6G  -385M
  15:03:07   27K   110G  108G     0     0  1.4G  -535M
  15:03:08   13K   108G  108G     0     0  2.2G  239M
  15:03:09   33K   107G  107G     0     0  1.3G  -639M
  15:03:10   16K   105G  102G     0     0  2.6G  704M
  15:03:11  7.2K   102G  102G     0     0  5.1G  3.1G
  15:03:12   42K   103G  102G     0     0  4.8G  2.8G
```

default output is 79 columns wide.  `arcsz` is changed to `size`, and `avail` is added:
```
$ arcstat 1
    time  read  miss  miss%  dmis  dm%  pmis  pm%  mmis  mm%  size     c  avail
15:12:30   965   241     24     0    0   241  100     1    0  115G  115G   2.0G
15:12:31  137K   34K     24     1    0   34K  100    34    0  115G  115G   2.2G
15:12:32  157K   39K     24     7    0   39K  100    43    0  115G  115G   1.9G
15:12:33  154K   38K     24     0    0   38K  100    37    0  115G  115G   2.1G
15:12:34  160K   39K     24     4    0   39K  100    42    0  115G  115G   2.3G
15:12:35  165K   41K     24     0    0   41K  100    41    0  115G  115G   2.0G
```

### Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [x] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

ahrens commented:
I think it would be useful to have `avail` be displayed by default, but unfortunately it makes the output 81 characters wide (when including the trailing space that's printed after each column).  I wasn't sure if we care about keeping it to 80 characters, or if folks see places that we could trim some width from the existing columns?

pzakha commented on man/man1/arcstat.1:
```diff
@@ -392,7 +392,17 @@ ARC reclaim needed
 .ad
 .RS 14n
-ARC free memory
+The ARC's idea of how much free memory there is, which includes evictable memory in the page cache.
```
As we've discussed offline, I think we should either add more explanations on what this stat means exactly, or refer to `avail` as the stat that most developers would likely be interested in.

ahrens replied:
Updated this to refer to `avail`.

pzakha commented on cmd/arcstat/arcstat:
```diff
@@ -101,6 +101,7 @@ cols = {
     "grow":       [4, 1000, "ARC grow disabled"],
     "need":       [4, 1024, "ARC reclaim need"],
     "free":       [4, 1024, "ARC free memory"],
+    "avail":       [4, 1024, "ARC available memory"],
```
nit: alignment

ahrens replied:
thanks for pointing that out.  fixed.

tonynguien commented:
> I think it would be useful to have `avail` be displayed by default, but unfortunately it makes the output 81 characters wide (when including the trailing space that's printed after each column). I wasn't sure if we care about keeping it to 80 characters, or if folks see places that we could trim some width from the existing columns?

Yes, it would be nice to have `avail` data by default. One idea is to rename `arcsz` to `size` which is the name used in `arcstats` kstat file.

ahrens commented:
OK, I added `avail` by default.  I also changed `arcsz` -> `size`, but left `arcsz` as a field for backwards compatibility in case scripts are using `-f arcsz`.

To make the default output fit in 80 columns, I eliminated the trailing `sep` (2 spaces) from each line, by using the more pythonic `str.join()`, which only adds the separator between elements, omitting the trailing separator.

Updated the first comment to reflect these changes.
